libs/sdk-core/src/breez_services.rs

```diff
@@ -483,23 +483,22 @@ impl BreezServices {
     /// See [SwapInfo] for details.
     pub async fn receive_onchain(
         &self,
-        _opening_fee_params: Option<OpeningFeeParams>,
+        opening_fee_params: Option<OpeningFeeParams>,
     ) -> Result<SwapInfo> {
-        let in_progress = self.in_progress_swap().await?;
-        if in_progress.is_some() {
+        if let Some(in_progress) = self.in_progress_swap().await? {
             return Err(anyhow!(format!(
                   "Swap in progress was detected for address {}.Use in_progress_swap method to get the current swap state",
-                  in_progress.unwrap().bitcoin_address
+                  in_progress.bitcoin_address
               )));
         }
         // TODO: Validate given opening_fee_params and use it if possible
         // We pick our own if None is supplied:
-        let longest_valid_opening_fee_params = get_longest_valid_opening_fee_params(
-            &mut self.lsp_info().await?.opening_fee_params_menu,
-        );
+        let channel_opening_fees = opening_fee_params.or(get_longest_valid_opening_fee_params(
+            &self.lsp_info().await?.opening_fee_params_menu,
+        )?);
         let swap_info = self
             .btc_receive_swapper
-            .create_swap_address(longest_valid_opening_fee_params)
+            .create_swap_address(channel_opening_fees)
```

Comment:
Does it make sense to give the `create_swap_address` method a request object as well?

Reply:
Since `create_swap_address` has only one arg (`channel_opening_fees`), IMO it makes little sense to wrap it in a separate request object.

Reply:
Yeah, I was just thinking this is a breaking change in the API structure. If we want to add another (optional) parameter, it will be a breaking change again.